drawing_agent/app/agent.py

```
@@ -85,23 +85,51 @@ async def _run_heavy_operations(self, image_base64: str) -> Dict[str, Any]:
         }
 
-    async def pre_analyze(self, path: str, page: int = 0) -> Dict:
+    async def pre_analyze(self, path: str, drawing_id: Optional[str] = None, page: int = 0) -> Dict:
         await self._ensure_initialized()
+
         # Используем lock, чтобы фоновый анализ не шел одновременно с вопросами
         async with self.lock:
             try:
                 loop = asyncio.get_running_loop()
+
+                # Если drawing_id не передан, генерируем его по хешу пути (для совместимости)
```

Ruff (0.15.12)

[warning] Comment contains ambiguous `е` (CYRILLIC SMALL LETTER IE). Did you mean `e` (LATIN SMALL LETTER E)?

(RUF003)